src/base/JSON.ml

```diff
@@ -213,10 +223,20 @@ and json_to_lit t v =
   | ADT (name, tlist) ->
       let vl = read_adt_json (get_id name) v tlist in
       vl
-  | _ ->
+  | PrimType _
+  | Address _ ->
       let tv = build_prim_lit_exn t (to_string_exn v) in
       tv
-
+  | FunType _
+  | TypeVar _
+  | PolyFun _
+  | Unit ->
+      let exn () =
+        mk_invalid_json ("Invalid type " ^ pp_typ t ^ " in JSON")
```

Comment:
Should we now name this function `json_to_lit_exn` (conforming to the other functions above that have `_exn` to mark that they throw exceptions).

Reply:
Will do.